src/SearchImpl/Indexing/CommitManager.cs

```diff
@@ -119,4 +121,58 @@ private void CommitWorker()
         }
 
     }
+
+    internal class NearRealTimeCommitManager : ICommitManager
+    {
+        private static readonly TimeSpan MaxWaitTime = TimeSpan.FromSeconds(10);
+        private DateTime _lastCommitTime;
+        private int _uncommittedActivityCount;
+
+        private static System.Timers.Timer _timer;
+        private static readonly int HearthBeatMilliseconds = 1000;
+
+        public void Start()
+        {
+            _lastCommitTime = DateTime.UtcNow;
+            _uncommittedActivityCount = 0;
+
+            _timer = new System.Timers.Timer(HearthBeatMilliseconds);
+            _timer.Elapsed += new System.Timers.ElapsedEventHandler(Timer_Elapsed);
+            _timer.Disposed += new EventHandler(Timer_Disposed);
+            _timer.Enabled = true;
+
+        }
+        private void Timer_Disposed(object sender, EventArgs e)
+        {
+            _timer.Elapsed -= new System.Timers.ElapsedEventHandler(Timer_Elapsed);
+            _timer.Disposed -= new EventHandler(Timer_Disposed);
+        }
+        private void Timer_Elapsed(object sender, ElapsedEventArgs e)
+        {
+            if (_uncommittedActivityCount > 0)
+                Commit();
+        }
+
+        public void ShutDown()
+        {
+            _timer.Enabled = false;
+            _timer.Dispose();
+        }
+
+        public void ActivityFinished()
+        {
+            Interlocked.Increment(ref _uncommittedActivityCount);
+
+            if (_uncommittedActivityCount == 1 || DateTime.UtcNow - _lastCommitTime > MaxWaitTime)
```

Comment:
Why  is the condition should be exactly _uncommittedActivityCount == 1?  What if it is 0?

Reply:
Thanks. This is a wrong condition.

Reply:
Zero is not a valid scenario. The _uncommittedActivityCount value cannot be negative and the line before "if" increments it.